### ISSUE TITLE
Missing lat/lon axes when converting from xr.DataArray to Cube

### ISSUE THREAD

**Issue by morganeoneill** [SciTools/iris#3714]
tl;dr I can convert a DataArray to a Cube, and I can successfully use the function `iris.analysis.cartography.rotate_pole` function with the corresponding lat and lon matrices to create local pole for my dataset. However I can't use `iris.analysis.cartography.rotate_winds` function on the cubes for `u` and `v` winds because the `x` and `y` dimensions are missing. It's weird because the `t` and `z` dimensions seem to be there just fine.

-----------------
I'm using a dataset from NASA's GEOS-5 model. I select a subsection of a massive netcdf dataset using xarray (I'll just show `u` wind here, I do the same with `v` wind):

    url = 'https://opendap.nccs.nasa.gov/dods/OSSE/G5NR/Ganymed/7km/0.0625_deg/inst/inst30mn_3d_U_Nv'
    dsu = xr.open_dataset(url)
    u = dsu['u'].sel(time=timesel,lat=latsel,lon=lonsel)
for my small slices of time, lat and lon (keeping all z levels)

This DataArray `dsu` has dimensions (1, 72, 129, 113) for dimension coordinates (time,lev,lat,lon). Then I convert to an Iris cube: 

    ucube = u.to_iris()

Everything looks like it's there (assume vcube is identical in shape/dimensions):

```
Input: print(ucube)

Output: eastward_wind  / (m s-1)            (time: 1; altitude: 72; latitude: 129; longitude: 113)
     Dimension coordinates:
          time                           x            -             -               -
          altitude                       -            x             -               -
          latitude                       -            -             x               -
          longitude                      -            -             -               x
```

It breaks for this function:

    newwinds = rotate_winds(ucube,vcube,cs)

With this traceback:

```
~/miniconda3/envs/secondenv/lib/python3.8/site-packages/iris/analysis/cartography.py in rotate_winds(u_cube, v_cube, target_cs)
   1031     msg = 'Coordinates differ between u and v cubes. Coordinate {!r} from ' \
   1032           'u cube does not equal coordinate {!r} from v cube.'
-> 1033     if u_cube.coord(axis='x') != v_cube.coord(axis='x'):
   1034         raise ValueError(msg.format(u_cube.coord(axis='x').name(),
   1035                                     v_cube.coord(axis='x').name()))

~/miniconda3/envs/secondenv/lib/python3.8/site-packages/iris/cube.py in coord(self, name_or_coord, standard_name, long_name, var_name, attributes, axis, contains_dimension, dimensions, coord_system, dim_coords)
   1495             msg = 'Expected to find exactly 1 %s coordinate, but found ' \
   1496                   'none.' % bad_name
-> 1497             raise iris.exceptions.CoordinateNotFoundError(msg)
   1498 
   1499         return coords[0]

CoordinateNotFoundError: 'Expected to find exactly 1  coordinate, but found none.'
```

Help! I can see using `ucube.coord(axis='t')` and `ucube.coord(axis='z')` yield the right dimensions, but an error is thrown (same as for the `rotate_winds` function, found no coordinate) for `x` and `y`.

I can't find any way around this, including the possibility of reassigning or redefining the axes in the cube. Is this a bug or am I being dense? Thanks!

**Comment by stephenworsley** [comment]
Hi! Thanks for the clear discription of your problem, I was able to replicate it without much of a problem.

It looks like you're correct about this failing because the longitude and latitude coordinates are not associated with the "x" and "y" axis. More specifically, this is due to how they behave with respect to the function `iris.util.guess_coord_axis()`. The logic for checking if a coordinate is associated with the X axis, for example, is as follows: https://github.com/SciTools/iris/blob/f5feb2831582e6b333923c3135cce4cca1d865a3/lib/iris/util.py#L250-L255
For your cube, I believe that the problem is that your latitude has a long_name of latitude/longitude but does not have a standard_name. Iris tends to consider the standard_name to be something which can change the functional properties of an object while the long_name is something which can change without affecting the behaviour of the object. You should be able to replace your coordinates with ones which contain standard names of latitude/longitude.
```
ulat = ucube.coord("latitude")
ulon = ucube.coord("longitude")

ulat.standard_name = "latitude"
ulon.standard_name = "longitude"

ucube.remove_coord("latitude")
ucube.add_dim_coord(ulat, 2)
ucube.remove_coord("longitude")
ucube.add_dim_coord(ulon, 3)
``` 

This should make it possible to use `rotate_winds`.

There may be a case for improving either xarray's `to_iris()` method or `iris.util.guess_coord_axis()`, though I'm not entirely sure what the consequences for doing so would be in either case.

**Comment by DPeterK** [comment]
Hi @morganeoneill - thanks indeed for opening this issue. It's really good to see the Xarray <--> Iris link in operation, although - as you've found - it's not perfect! What you've hit here is effectively a metadata translation issue between how the Xarray and Iris data models are defined, meaning that Xarray does not provide all the metadata that an Iris cube needs to be fully functional when you convert an Xarray dataset to an Iris cube. 

As such, it would be worth also raising this issue on the [Xarray issue tracker](https://github.com/pydata/xarray/issues) so that the Xarray devs are also aware of this - although this would be a great opportunity for some of the Iris devs to jump into the Xarray codebase to provide the fix for this issue! (Anyone?)

Note that you can use Iris to load your dataset directly:

```python
url = 'https://opendap.nccs.nasa.gov/dods/OSSE/G5NR/Ganymed/7km/0.0625_deg/inst/inst30mn_3d_U_Nv'
cube = iris.load_cube(url)
print(cube)
```
```
eastward_wind  / (m s-1)            (time: 36576; altitude: 72; latitude: 2881; longitude: 5760)
     Dimension coordinates:
          time                           x                -             -                -
          altitude                       -                x             -                -
          latitude                       -                -             x                -
          longitude                      -                -             -                x
     Attributes:
          Conventions: COARDS
GrADS
          dataType: Grid
          extra_das_attribute: This is an example of metadata added using a supplementary DAS file. See...
          history: Fri Feb 21 04:11:14 EST 2020 : imported by GrADS Data Server 2.0
          title: 3d,30-Minute,Instantaneous,Model-Level,Full Resolution,Zonal Wind
```

The Iris-only representation includes the missing metadata needed for the `rotate_winds` command:
```python
print(cube.coords(axis='x'))
```
```
[DimCoord(array([-180.    , -179.9375, -179.875 , ...,  179.8125,  179.875 ,
        179.9375]), standard_name='longitude', units=Unit('degrees'), long_name='longitude', var_name='lon', attributes={'grads_dim': 'x', 'grads_mapping': 'linear', 'grads_size': '5760', 'maximum': 179.9375, 'minimum': -180.0, 'resolution': 0.0625})]
```

To subset the cube in Iris (the Iris equivalent of Xarray's `ds['key'].sel()`, you can either index the cube directly like a NumPy array:
```python
subcube = cube[0, :, 100:200, 100:200]
```
or use a [constraint](https://scitools.org.uk/iris/docs/latest/iris/iris.html?#iris.Constraint) to subset the cube based on coordinate values. Constraints are described in more detail in the [Iris teaching course](https://github.com/SciTools/courses/blob/master/course_content/iris_course/3.Subcube_Extraction.ipynb), §3.2.

Hope this helps!

**Comment by morganeoneill** [comment]
Thanks so much @stephenworsley and @DPeterK! That solution worked great and I'm off to the races. I might also switch to loading directly into Iris, but I need metpy too and I haven't yet explored the iris/metpy connection....big world out there! Thanks a lot, I really appreciate your help!
